src/transactions.rs

```diff
@@ -368,7 +368,13 @@ impl<L: Ledger + Serialize + DeserializeOwned> Transactions<L> {
 
     /// Remove a transaction from the pending index when it is known to have timed out
     pub async fn remove_expired(&mut self, timeout: u64) -> Result<(), KeystoreError<L>> {
-        if let Some(expiring_uids) = self.expiring_txns.index.get_mut(&timeout).cloned() {
+        if let Some(expiring_uids) = self
+            .expiring_txns
+            .index()
+            .clone()
+            .get_mut(&timeout)
+            .cloned()
```

Comment:
This seems like a smell. The first clone is necessary to call `get_mut` on a const reference. But then we are cloning the resulting mutable reference again? Do we need `get_mut` here at all? Or would `.index().get(&timeout).cloned()` work?

Reply:
Yes, it works!